js/opening-hours.functions.js

```diff
@@ -43,9 +43,338 @@ var getClosest = function (elem, selector) {
+  if (this.type !== null) {
+    return this.type;
+  }
+
+  if (typeof this.element.dataset.type === 'undefined') {
+    this.printError('Please provide a widget type.');
+    this.type = false;
+  }
+  else {
+    this.type = this.element.dataset.type;
+  }
+
+  return this.type;
+};
+
+/**
+ * Get the item Service ID.
+ *
+ * @return {false|integer}
+ *   The Service id (if any).
+ */
+OpeningHoursItem.prototype.getService = function () {
+  if (this.service !== null) {
+    return this.service;
+  }
+
+  if (isNaN(this.element.dataset.service)) {
+    this.printError('Please provide a service id.');
+    this.service = false;
+  }
+  else {
+    this.service = parseInt(this.element.dataset.service);
+  }
+
+  return this.service;
+};
+
+/**
+ * Get the item Channel ID.
+ *
+ * @return {false|integer}
+ *   The channel (if any).
+ */
+OpeningHoursItem.prototype.getChannel = function () {
+  if (this.channel !== null) {
+    return this.channel;
+  }
+
+  if (isNaN(this.element.dataset.channel)) {
+    this.printError('Please provide a channel id.');
+    this.channel = false;
+  }
+  else {
+    this.channel = parseInt(this.element.dataset.channel);
+  }
+
+  return this.channel;
+};
+
+/**
+ * Get the date the item wants to show the widget for.
+ *
+ * @return {OpeningHoursWidgetDate}
+ *   The Opening Hours widget date object.
+ */
+OpeningHoursItem.prototype.getDate = function () {
+  if (this.date !== null) {
+    return this.date;
+  }
+
+  // If the options request date is set, use that one (oh_date request param).
+  if (this.options.requestDate) {
+    this.date = new OpeningHoursWidgetDate(new Date(this.options.requestDate));
+    return this.date;
+  }
+
+  // Fallback to current date if no date is set in the widget data (data-date).
+  if (typeof this.element.dataset.date === 'undefined') {
+    this.date = new OpeningHoursWidgetDate(new Date());
+    return this.date;
+  }
+
+  // Date from the data-date property.
+  this.date = new OpeningHoursWidgetDate(new Date(this.element.dataset.date));
+  return this.date;
+};
+
+/**
+ * Get the period filter the item wants to limit the opening hours by.
+ *
+ * @returns {OpeningHoursPeriod}
+ *   The period object.
+ */
+OpeningHoursItem.prototype.getPeriod = function () {
+  if (this.period !== null) {
+    return this.period;
+  }
+
+  var from = typeof this.element.dataset.from !== 'undefined'
+    ? new Date(this.element.dataset.from)
+    : null;
+  var until = typeof this.element.dataset.until !== 'undefined'
+    ? new Date(this.element.dataset.until)
+    : null;
+
+  this.period = new OpeningHoursPeriod(from, until);
+  return this.period;
+};
+
+/**
+ * Has the widget everything to create a widget of it.
+ *
+ * @return {boolean}
+ *   Everything in place to create a widget out of it.
+ */
+OpeningHoursItem.prototype.isWidgetable = function () {
+  return this.getType() !== false
+    && this.getService() !== false
+    && this.getChannel() !== false;
```

Comment:
There are use cases where the channel is not defined, which results in fetching the openinghours for all channels from the API, so I suppose that `this.getChannel() !== false` can be removed

Reply:
Fixed.